### ISSUE TITLE
RE( scan_nd ) does not work properly

### ISSUE THREAD

**Issue by yugangzhang** [NSLS-II/Bug-Reports#198]
We have a problem to run RE( scan_nd ) in the new env:  /opt/conda_envs/collection-2019-1.2-chx/

Code:
python```
	inner = cycler(dcm.b,th_B)+cycler(ivu_gap,gap)	
	plan = scan_nd(det,inner)	 
	RE(plan)
```
By debug, we got inner=
```
Out[59]: (cycler(EpicsMotor(prefix='XF:11IDA-OP{Mono:DCM-Ax:B}Mtr', name='dcm_b', parent='dcm', settle_time=0.0, timeout=None, read_attrs=['user_readback', 'user_setpoint'], configuration_attrs=['user_offset', 'user_offset_dir', 'velocity', 'acceleration', 'motor_egu']), [-14.135709002929746, -14.117917253207015]) + cycler(Undulator(prefix='SR:C11-ID:G1{IVU20:1', name='ivu_gap', settle_time=0.0, timeout=None, read_attrs=['readback', 'setpoint', 'actuate', 'stop_signal'], configuration_attrs=[], limits=None, egu=''), [5.849359983576296, 5.854505845881992]))
```
Error:
python```

Transient Scan ID: 2645     Time: 2019/01/28 12:04:18
Persistent Unique Scan ID: 'a7ffd082-88ea-42fc-aeef-f03aae8d33e9'
Transient Scan ID: 2645 @ 2019/01/28 12:04:18
Persistent Unique Scan ID: 'a7ffd082-88ea-42fc-aeef-f03aae8d33e9'
.ipython/profile_collection/startup/99-bluesky.py:117: UserWarning: This olog is giving errors. This will not be logged.Error:'dict object' has no attribute 'start'
  inner = cycler(dcm.b,th_B)+cycler(ivu_gap,gap)
New stream: 'primary'
+-----------+------------+------------+------------+------------------------+
|   seq_num |       time |    ivu_gap |      dcm_b | xray_eye1_stats1_total |
+-----------+------------+------------+------------+------------------------+
+-----------+------------+------------+------------+------------------------+
generator scan_nd ['a7ffd082'] (scan num: 2645)



The scan ID is: 2645
The scan ID is: 2645
The scan ID is: 2645
[E 12:04:19.852 run_engine:1264] Run aborted
    Traceback (most recent call last):
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py", line 1153, in _run
        msg = self._plan_stack[-1].send(resp)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 1294, in __call__
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 1147, in baseline_wrapper
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 790, in monitor_during_wrapper
        return (yield from plan2)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 169, in plan_mutator
        raise ex
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 122, in plan_mutator
        msg = plan_stack[-1].send(ret)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 169, in plan_mutator
        raise ex
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 122, in plan_mutator
        msg = plan_stack[-1].send(ret)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 848, in fly_during_wrapper
        return (yield from plan2)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 169, in plan_mutator
        raise ex
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 122, in plan_mutator
        msg = plan_stack[-1].send(ret)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 169, in plan_mutator
        raise ex
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 122, in plan_mutator
        msg = plan_stack[-1].send(ret)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plans.py", line 750, in scan_nd
        return (yield from inner_scan_nd())
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py", line 1011, in dec_inner
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 939, in stage_wrapper
        return (yield from finalize_wrapper(inner(), unstage_devices()))
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 500, in finalize_wrapper
        ret = yield from plan
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 937, in inner
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py", line 1011, in dec_inner
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 326, in run_wrapper
        else_plan=close_run)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 562, in contingency_wrapper
        ret = yield from plan
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plans.py", line 748, in inner_scan_nd
        yield from per_step(detectors, step, pos_cache)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py", line 952, in one_nd_step
        yield from trigger_and_read(list(detectors) + list(motors))
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py", line 802, in trigger_and_read
        rewindable))
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 680, in rewindable_wrapper
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py", line 798, in inner_trigger_and_read
        yield from save()
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py", line 57, in save
        return (yield Msg('save'))
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 194, in plan_mutator
        inner_ret = yield msg
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 194, in plan_mutator
        inner_ret = yield msg
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 194, in plan_mutator
        inner_ret = yield msg
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py", line 1211, in _run
        new_response = yield from coro(msg)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py", line 1687, in _save
        yield from self.emit(DocumentNames.descriptor, doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/asyncio/coroutines.py", line 212, in coro
        res = func(*args, **kw)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py", line 2324, in emit
        self.dispatcher.process(name, doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py", line 2344, in process
        exceptions = self.cb_registry.process(name, name.name, doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py", line 334, in process
        func(*args, **kwargs)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py", line 415, in __call__
        return mtd(*args, **kwargs)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/best_effort.py", line 93, in __call__
        super().__call__(name, doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/core.py", line 17, in __call__
        return getattr(self, name)(doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/best_effort.py", line 341, in descriptor
        live_scatter('start', self._start_doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/core.py", line 17, in __call__
        return getattr(self, name)(doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/mpl_plotting.py", line 225, in start
        norm=self._norm, cmap=self.cmap, **self.kwargs)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/matplotlib/__init__.py", line 1805, in inner
        return func(ax, *args, **kwargs)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/matplotlib/axes/_axes.py", line 4195, in scatter
        isinstance(c[0], str))):
    IndexError: list index out of range
[E 12:04:19.852 run_engine:1264] Run aborted
    Traceback (most recent call last):
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py", line 1153, in _run
        msg = self._plan_stack[-1].send(resp)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 1294, in __call__
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 1147, in baseline_wrapper
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 790, in monitor_during_wrapper
        return (yield from plan2)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 169, in plan_mutator
        raise ex
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 122, in plan_mutator
        msg = plan_stack[-1].send(ret)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 169, in plan_mutator
        raise ex
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 122, in plan_mutator
        msg = plan_stack[-1].send(ret)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 848, in fly_during_wrapper
        return (yield from plan2)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 169, in plan_mutator
        raise ex
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 122, in plan_mutator
        msg = plan_stack[-1].send(ret)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 169, in plan_mutator
        raise ex
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 122, in plan_mutator
        msg = plan_stack[-1].send(ret)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plans.py", line 750, in scan_nd
        return (yield from inner_scan_nd())
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py", line 1011, in dec_inner
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 939, in stage_wrapper
        return (yield from finalize_wrapper(inner(), unstage_devices()))
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 500, in finalize_wrapper
        ret = yield from plan
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 937, in inner
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py", line 1011, in dec_inner
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 326, in run_wrapper
        else_plan=close_run)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 562, in contingency_wrapper
        ret = yield from plan
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plans.py", line 748, in inner_scan_nd
        yield from per_step(detectors, step, pos_cache)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py", line 952, in one_nd_step
        yield from trigger_and_read(list(detectors) + list(motors))
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py", line 802, in trigger_and_read
        rewindable))
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 680, in rewindable_wrapper
        return (yield from plan)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py", line 798, in inner_trigger_and_read
        yield from save()
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py", line 57, in save
        return (yield Msg('save'))
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 194, in plan_mutator
        inner_ret = yield msg
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 194, in plan_mutator
        inner_ret = yield msg
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py", line 194, in plan_mutator
        inner_ret = yield msg
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py", line 1211, in _run
        new_response = yield from coro(msg)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py", line 1687, in _save
        yield from self.emit(DocumentNames.descriptor, doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/asyncio/coroutines.py", line 212, in coro
        res = func(*args, **kw)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py", line 2324, in emit
        self.dispatcher.process(name, doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py", line 2344, in process
        exceptions = self.cb_registry.process(name, name.name, doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py", line 334, in process
        func(*args, **kwargs)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py", line 415, in __call__
        return mtd(*args, **kwargs)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/best_effort.py", line 93, in __call__
        super().__call__(name, doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/core.py", line 17, in __call__
        return getattr(self, name)(doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/best_effort.py", line 341, in descriptor
        live_scatter('start', self._start_doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/core.py", line 17, in __call__
        return getattr(self, name)(doc)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/mpl_plotting.py", line 225, in start
        norm=self._norm, cmap=self.cmap, **self.kwargs)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/matplotlib/__init__.py", line 1805, in inner
        return func(ax, *args, **kwargs)
      File "/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/matplotlib/axes/_axes.py", line 4195, in scatter
        isinstance(c[0], str))):
    IndexError: list index out of range
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
~/.ipython/profile_collection/startup/99-bluesky.py in <module>
----> 1 RE(p)

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py in __call__(self, *args, **metadata_kw)
    754                     # it (unless it is a canceled error)
    755                     if exc is not None:
--> 756                         raise exc
    757 
    758             if self._interrupted:

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py in _run(self)
   1263             self._reason = str(err)
   1264             self.log.exception("Run aborted")
-> 1265             raise err
   1266         finally:
   1267             # Some done_callbacks may still be alive in other threads.

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py in _run(self)
   1151                     else:
   1152                         try:
-> 1153                             msg = self._plan_stack[-1].send(resp)
   1154                         # We have exhausted the top generator
   1155                         except StopIteration:

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in __call__(self, plan)
   1292         plan = monitor_during_wrapper(plan, self.monitors)
   1293         plan = baseline_wrapper(plan, self.baseline)
-> 1294         return (yield from plan)

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in baseline_wrapper(plan, devices, name)
   1145     if not devices:
   1146         # no-op
-> 1147         return (yield from plan)
   1148     else:
   1149         return (yield from plan_mutator(plan, insert_baseline))

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in monitor_during_wrapper(plan, signals)
    788     plan1 = plan_mutator(plan, insert_after_open)
    789     plan2 = plan_mutator(plan1, insert_before_close)
--> 790     return (yield from plan2)
    791 
    792 

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    167                     continue
    168                 else:
--> 169                     raise ex
    170         # if inserting / mutating, put new generator on the stack
    171         # and replace the current msg with the first element from the

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    120             ret = result_stack.pop()
    121             try:
--> 122                 msg = plan_stack[-1].send(ret)
    123             except StopIteration as e:
    124                 # discard the exhausted generator

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    167                     continue
    168                 else:
--> 169                     raise ex
    170         # if inserting / mutating, put new generator on the stack
    171         # and replace the current msg with the first element from the

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    120             ret = result_stack.pop()
    121             try:
--> 122                 msg = plan_stack[-1].send(ret)
    123             except StopIteration as e:
    124                 # discard the exhausted generator

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in fly_during_wrapper(plan, flyers)
    846     plan1 = plan_mutator(plan, insert_after_open)
    847     plan2 = plan_mutator(plan1, insert_before_close)
--> 848     return (yield from plan2)
    849 
    850 

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    167                     continue
    168                 else:
--> 169                     raise ex
    170         # if inserting / mutating, put new generator on the stack
    171         # and replace the current msg with the first element from the

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    120             ret = result_stack.pop()
    121             try:
--> 122                 msg = plan_stack[-1].send(ret)
    123             except StopIteration as e:
    124                 # discard the exhausted generator

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    167                     continue
    168                 else:
--> 169                     raise ex
    170         # if inserting / mutating, put new generator on the stack
    171         # and replace the current msg with the first element from the

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    120             ret = result_stack.pop()
    121             try:
--> 122                 msg = plan_stack[-1].send(ret)
    123             except StopIteration as e:
    124                 # discard the exhausted generator

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plans.py in scan_nd(detectors, cycler, per_step, md)
    748             yield from per_step(detectors, step, pos_cache)
    749 
--> 750     return (yield from inner_scan_nd())
    751 
    752 

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py in dec_inner(*inner_args, **inner_kwargs)
   1009                 plan = gen_func(*inner_args, **inner_kwargs)
   1010                 plan = wrapper(plan, *args, **kwargs)
-> 1011                 return (yield from plan)
   1012             return dec_inner
   1013         return dec

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in stage_wrapper(plan, devices)
    937         return (yield from plan)
    938 
--> 939     return (yield from finalize_wrapper(inner(), unstage_devices()))
    940 
    941 

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in finalize_wrapper(plan, final_plan, pause_for_debug)
    498     cleanup = True
    499     try:
--> 500         ret = yield from plan
    501     except GeneratorExit:
    502         cleanup = False

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in inner()
    935     def inner():
    936         yield from stage_devices()
--> 937         return (yield from plan)
    938 
    939     return (yield from finalize_wrapper(inner(), unstage_devices()))

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py in dec_inner(*inner_args, **inner_kwargs)
   1009                 plan = gen_func(*inner_args, **inner_kwargs)
   1010                 plan = wrapper(plan, *args, **kwargs)
-> 1011                 return (yield from plan)
   1012             return dec_inner
   1013         return dec

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in run_wrapper(plan, md)
    324     yield from contingency_wrapper(plan,
    325                                    except_plan=except_plan,
--> 326                                    else_plan=close_run)
    327     return rs_uid
    328 

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in contingency_wrapper(plan, except_plan, else_plan, final_plan, pause_for_debug)
    560     cleanup = True
    561     try:
--> 562         ret = yield from plan
    563     except GeneratorExit:
    564         cleanup = False

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plans.py in inner_scan_nd()
    746     def inner_scan_nd():
    747         for step in list(cycler):
--> 748             yield from per_step(detectors, step, pos_cache)
    749 
    750     return (yield from inner_scan_nd())

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py in one_nd_step(detectors, step, pos_cache)
    950     motors = step.keys()
    951     yield from move_per_step(step, pos_cache)
--> 952     yield from trigger_and_read(list(detectors) + list(motors))
    953 
    954 

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py in trigger_and_read(devices, name)
    800     from .preprocessors import rewindable_wrapper
    801     return (yield from rewindable_wrapper(inner_trigger_and_read(),
--> 802                                           rewindable))
    803 
    804 

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in rewindable_wrapper(plan, rewindable)
    678                                             restore_rewindable()))
    679     else:
--> 680         return (yield from plan)
    681 
    682 

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py in inner_trigger_and_read()
    796             if reading is not None:
    797                 ret.update(reading)
--> 798         yield from save()
    799         return ret
    800     from .preprocessors import rewindable_wrapper

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/plan_stubs.py in save()
     55     :func:`bluesky.plan_stubs.create`
     56     """
---> 57     return (yield Msg('save'))
     58 
     59 

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    192         try:
    193             # yield out the 'current message' and collect the return
--> 194             inner_ret = yield msg
    195         except GeneratorExit:
    196             # special case GeneratorExit.  We must clean up all of our plans

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    192         try:
    193             # yield out the 'current message' and collect the return
--> 194             inner_ret = yield msg
    195         except GeneratorExit:
    196             # special case GeneratorExit.  We must clean up all of our plans

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    192         try:
    193             # yield out the 'current message' and collect the return
--> 194             inner_ret = yield msg
    195         except GeneratorExit:
    196             # special case GeneratorExit.  We must clean up all of our plans

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/preprocessors.py in plan_mutator(plan, msg_proc)
    192         try:
    193             # yield out the 'current message' and collect the return
--> 194             inner_ret = yield msg
    195         except GeneratorExit:
    196             # special case GeneratorExit.  We must clean up all of our plans

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py in _run(self)
   1209                         # exceptions (coming in via throw) can be
   1210                         # raised
-> 1211                         new_response = yield from coro(msg)
   1212 
   1213                     # special case `CancelledError` and let the outer

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py in _save(self, msg)
   1685                        configuration=config, name=desc_key,
   1686                        hints=hints, object_keys=object_keys)
-> 1687             yield from self.emit(DocumentNames.descriptor, doc)
   1688             self.log.debug("Emitted Event Descriptor with name %r containing "
   1689                            "data keys %r (uid=%r)", desc_key,

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/asyncio/coroutines.py in coro(*args, **kw)
    210         @functools.wraps(func)
    211         def coro(*args, **kw):
--> 212             res = func(*args, **kw)
    213             if (base_futures.isfuture(res) or inspect.isgenerator(res) or
    214                 isinstance(res, CoroWrapper)):

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py in emit(self, name, doc)
   2322         "Process blocking callbacks and schedule non-blocking callbacks."
   2323         _validate(doc, schemas[name])
-> 2324         self.dispatcher.process(name, doc)
   2325 
   2326 

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/run_engine.py in process(self, name, doc)
   2342         doc : dict
   2343         """
-> 2344         exceptions = self.cb_registry.process(name, name.name, doc)
   2345         for exc, traceback in exceptions:
   2346             warn("A %r was raised during the processing of a %s "

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py in process(self, sig, *args, **kwargs)
    332             for cid, func in list(self.callbacks[sig].items()):
    333                 try:
--> 334                     func(*args, **kwargs)
    335                 except ReferenceError:
    336                     self._remove_proxy(func)

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/utils.py in __call__(self, *args, **kwargs)
    413             mtd = self.func
    414         # invoke the callable and return the result
--> 415         return mtd(*args, **kwargs)
    416 
    417     def __eq__(self, other):

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/best_effort.py in __call__(self, name, doc)
     91             return
     92 
---> 93         super().__call__(name, doc)
     94 
     95     def start(self, doc):

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/core.py in __call__(self, name, doc)
     15     def __call__(self, name, doc):
     16         "Dispatch to methods expecting particular doc types."
---> 17         return getattr(self, name)(doc)
     18 
     19     def event(self, doc):

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/best_effort.py in descriptor(self, doc)
    339                                                # Let clim autoscale.
    340                                                ax=ax)
--> 341                     live_scatter('start', self._start_doc)
    342                     live_scatter('descriptor', doc)
    343                     self._live_scatters[doc['uid']][I_key] = live_scatter

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/core.py in __call__(self, name, doc)
     15     def __call__(self, name, doc):
     16         "Dispatch to methods expecting particular doc types."
---> 17         return getattr(self, name)(doc)
     18 
     19     def event(self, doc):

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/bluesky/callbacks/mpl_plotting.py in start(self, doc)
    223         self._Idata.clear()
    224         sc = self.ax.scatter(self._xdata, self._ydata, c=self._Idata,
--> 225                              norm=self._norm, cmap=self.cmap, **self.kwargs)
    226         self._sc.append(sc)
    227         self.sc = sc

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/matplotlib/__init__.py in inner(ax, data, *args, **kwargs)
   1803                         "the Matplotlib list!)" % (label_namer, func.__name__),
   1804                         RuntimeWarning, stacklevel=2)
-> 1805             return func(ax, *args, **kwargs)
   1806 
   1807         inner.__doc__ = _add_data_doc(inner.__doc__,

/opt/conda_envs/collection-2019-1.2-chx/lib/python3.6/site-packages/matplotlib/axes/_axes.py in scatter(self, x, y, s, c, marker, cmap, norm, vmin, vmax, alpha, linewidths, verts, edgecolors, **kwargs)
   4193                 isinstance(c, str) or
   4194                 (isinstance(c, collections.Iterable) and
-> 4195                     isinstance(c[0], str))):
   4196             c_array = None
   4197         else:

IndexError: list index out of range

```
Your help will be greatly appreciated.

**Comment by tacaswell** [comment]
This is a Matplotlib bug which is fixed upstream, in process of updating the enviroments.

**Comment by mrakitin** [comment]
Observed this at ISS as well. Fixed by installing matplotlib 3.0.3 from PyPI. This version of matplotlib will be build in our conda channel soon and will be deployed on the floor during the upcoming deployment.

**Comment by tacaswell** [comment]
Closing as we understand this.
